datafusion/proto/src/from_proto.rs

```diff
@@ -581,12 +575,30 @@ impl TryFrom<&protobuf::scalar_value::Value> for ScalarValue {
                 )
             }
             Value::Date64Value(v) => ScalarValue::Date64(Some(*v)),
-            Value::TimeSecondValue(v) => ScalarValue::TimestampSecond(Some(*v), None),
-            Value::TimeMillisecondValue(v) => {
-                ScalarValue::TimestampMillisecond(Some(*v), None)
-            }
             Value::IntervalYearmonthValue(v) => ScalarValue::IntervalYearMonth(Some(*v)),
             Value::IntervalDaytimeValue(v) => ScalarValue::IntervalDayTime(Some(*v)),
+            Value::TimestampValue(v) => {
+                let ts_value = v.value.as_ref().ok_or(Error::required("value"))?;
+                let timezone = if v.timezone.is_empty() {
+                    None
+                } else {
+                    Some(v.timezone.clone())
+                };
```

Comment:
Maybe it could be written like this:
```suggestion
                let timezone = v.timezone.map(|v| Some(v.clone()))?
```